azkaban-common/src/main/java/azkaban/project/DirectoryYamlFlowLoader.java

```diff
@@ -45,7 +45,17 @@
  */
 public class DirectoryYamlFlowLoader implements FlowLoader {
 
+  // Pattern to match job variables in condition expressions: ${jobName:variable}
+  public static final Pattern CONDITION_VARIABLE_REPLACEMENT_PATTERN = Pattern
+      .compile("\\$\\{([^:{}]+):([^:{}]+)\\}");
   private static final Logger logger = LoggerFactory.getLogger(DirectoryYamlFlowLoader.class);
+  // Pattern to match conditionOnJobStatus macros, e.g. one_success, all_done
+  private static final Pattern CONDITION_ON_JOB_STATUS_PATTERN =
+      Pattern.compile("(?i)\\b(" + StringUtils.join(ConditionOnJobStatus.values(), "|") + ")\\b");
+  // Pattern to match a number or a string, e.g. 1234, "hello", 'foo'
+  private static final Pattern DIGIT_STRING_PATTERN = Pattern.compile("\\d+|'.*'|\".*\"");
```

Comment:
is this private or public? CONDITION_VARIABLE_REPLACEMENT_PATTERN is public

Reply:
CONDITION_VARIABLE_REPLACEMENT_PATTERN is public because it's also used in another file as well. DIGIT_STRING_PATTERN is private and only used by this file.